index.js

```diff
@@ -20,31 +20,38 @@ module.exports = function(options) {
     options.watch(client);
   }
 
-  return through.obj(function transform(file, enc, callback) {
-    if (file.isStream()) {
-      return callback(new PluginError('gulp-scp2', 'Streaming not supported.'));
-    }
-
-    var path = join(options.dest, file.relative);
-    client.mkdir(dirname(path), function(err) {
-      if (err) {
-        return callback(new PluginError('gulp-scp2', err));
+  return through.obj(function(file, enc, callback) {
+      if (file.isStream()) {
+        return callback(new PluginError('gulp-scp2', 'Streaming not supported.'));
       }
 
-      client.write({
-        destination: path,
-        content: file.contents
-      }, function(err) {
+      var filePath = pathLib.relative(process.cwd(), file.path);
```

Comment:
Ok forget the pull request your code does not generate folder structure in the destination. I will write my own. 



> On May 30, 2015, at 2:16 PM, Haoliang Gao <notifications@github.com> wrote:
> 
> In index.js:
> 
> >        }
> >  
> > -      client.write({
> > -        destination: path,
> > -        content: file.contents
> > -      }, function(err) {
> > +      var filePath = pathLib.relative(process.cwd(), file.path);
> You can use glob for that,
> 
> gulp.src(['**.js', '**.json']);
> ** stand for everything containing /
> 
> —
> Reply to this email directly or view it on GitHub.
>